Sources/Commands/PackageTools/PluginCommand.swift

```diff
@@ -160,22 +160,14 @@ struct PluginCommand: SwiftCommand {
             + getEnvSearchPaths(pathString: ProcessEnv.path, currentWorkingDirectory: .none)
 
         // Build or bring up-to-date any executable host-side tools on which this plugin depends. Add them and any binary dependencies to the tool-names-to-path map.
-        var toolNamesToPaths: [String: AbsolutePath] = [:]
-        // Add supported triples info per tool so they can be looked up when running the tool
-        var toolNamesToTriples: [String: [String]] = [:]
-        for dep in try plugin.accessibleTools(packageGraph: packageGraph, fileSystem: swiftTool.fileSystem, environment: try swiftTool.buildParameters().buildEnvironment, for: try pluginScriptRunner.hostTriple) {
+        let (toolNamesToPaths, toolNamesToTriples) = try plugin.processAccessibleTools(packageGraph: packageGraph, fileSystem: swiftTool.fileSystem, environment: try swiftTool.buildParameters().buildEnvironment, for: try pluginScriptRunner.hostTriple) { name, path in
             let buildSystem = try swiftTool.createBuildSystem(explicitBuildSystem: .native, cacheBuildManifest: false)
```

Comment:
I am preserving the previous behavior here (build system gets created per tool), but this does seem a little odd to me. @abertelrud do you remember if there was a reason for this or maybe just something that snuck in during previous refactoring? I think it would make sense to create the build system once, outside of the loop.

Reply:
It doesn't sound intentional; at least, I have no memory of it being intentional.  So having a single build system seems good to me.

Reply:
Great, I'll update that, but will do so in another PR, just to keep this refactoring separate from a technically functional change.

Reply:
https://github.com/apple/swift-package-manager/pull/5974